From 460ddd2729f0dbb2723e48f3a22ffccbb78b42c7 Mon Sep 17 00:00:00 2001
From: Dave Stevenson <dave.stevenson@raspberrypi.com>
Date: Tue, 31 Mar 2020 17:54:08 +0100
Subject: [PATCH] dt: Drop I2C for Pi4 HDMI interfaces to 97.5kHz.

It was set to 390kHz, which is outside of the required spec for
reading HDMI (max 100kHz). The i2c-brcmstb driver only supports
a number of fixed bus speeds, of which 97.5kHz is the closest to
100kHz without exceeding it.

Signed-off-by: Dave Stevenson <dave.stevenson@raspberrypi.com>
---
 arch/arm/boot/dts/bcm2711.dtsi | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- a/arch/arm/boot/dts/bcm2711.dtsi
+++ b/arch/arm/boot/dts/bcm2711.dtsi
@@ -358,7 +358,7 @@
 			compatible = "brcm,bcm2711-hdmi-i2c";
 			reg = <0x7ef04500 0x100>, <0x7ef00b00 0x300>;
 			reg-names = "bsc", "auto-i2c";
-			clock-frequency = <390000>;
+			clock-frequency = <97500>;
 			status = "disabled";
 		};
 
@@ -395,7 +395,7 @@
 			compatible = "brcm,bcm2711-hdmi-i2c";
 			reg = <0x7ef09500 0x100>, <0x7ef05b00 0x300>;
 			reg-names = "bsc", "auto-i2c";
-			clock-frequency = <390000>;
+			clock-frequency = <97500>;
 			status = "disabled";
 		};
 	};
